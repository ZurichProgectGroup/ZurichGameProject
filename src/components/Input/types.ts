import { FC, InputHTMLAttributes } from 'react';

export type OwnProps = {
    errorText?: string;
    labelText: string,
    description?: string,
} & InputHTMLAttributes<HTMLInputElement>;

export type Props = FC<OwnProps>;
