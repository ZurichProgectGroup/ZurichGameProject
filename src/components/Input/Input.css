.input {
    display: flex;
    flex-direction: column;
}

.input__label {
    margin-bottom: 10px;
    color: var(--font-color);
    text-transform: uppercase;
}

.input__field {
    color: var(--font-color);
    background: transparent;
    border: 1px solid var(--main);
    box-shadow: 0 0 5px var(--main);
    border-radius: 4px;
    padding: 10px;
    transition: all 0.2s;

    &:focus {
        outline: none;
        box-shadow: 0 0 5px var(--focus);
        border-color: var(--focus);
    }
}

.input__desc {
    margin-top: 6px;
    text-transform: uppercase;
    font-size: 12px;
    min-height: 14px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;

    &_hidden {
        visibility: hidden;
    }
}

.input_warn {
    .input__field {
        box-shadow: 0 0 5px var(--warn);
        border-color: var(--warn);
    }

    .input__desc {
        color: var(--warn);
    }
}
