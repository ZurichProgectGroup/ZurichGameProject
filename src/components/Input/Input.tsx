import React, { useMemo } from 'react';
import cn from 'classnames';
import './Input.css';
import type { Props } from './types';

const Input = (props: Props) => {
    const {
        errorText,
        labelText,
        description,
        className,
        ...inputProps
    } = props;

    const actualDescription = useMemo(() => description || errorText, [description, errorText]);

    return (
        <label className={cn('input', className, {
            input_warn: Boolean(errorText),
        })}
        >
            <span className="input__label">{labelText}</span>
            <input className="input__field" {...inputProps} />
            <span className={cn('input__desc', {
                input__desc_hidden: !actualDescription,
            })}
            >
                {actualDescription}
            </span>
        </label>
    );
};

export default Input;
