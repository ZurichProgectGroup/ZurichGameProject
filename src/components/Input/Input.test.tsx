import React from 'react';
import { render } from '@testing-library/react';
import Input from './Input';

it('renders correctly', () => {
    const { asFragment } = render(<Input labelText="test input" />);

    expect(asFragment()).toMatchSnapshot();
});
