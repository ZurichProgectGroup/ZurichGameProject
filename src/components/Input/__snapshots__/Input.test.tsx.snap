// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <label
    class="input"
  >
    <span
      class="input__label"
    >
      test input
    </span>
    <input
      class="input__field"
    />
    <span
      class="input__desc input__desc_hidden"
    />
  </label>
</DocumentFragment>
`;
